public/scripts/play.js

```diff
@@ -210,6 +215,21 @@ function movePlayer(e) {
       // no need to update server if player didn't move
       return;
   }
+  if (e.key === currentKey) {
+    if (!delay) {
+      walkingSound.play();
+      delay = 10;
```

Comment:
Also a bit confused about currentKey? Why do you need to do this check? Wouldn't it be enough to onKeyDown do sound.play and onKeyUp sound.pause/stop?

Reply:
To give you maybe a bit more simple idea. Audio has a property of loop so when you set it to true it will loop. Another property is paused. So you can check whether the audio is paused and only then play it instead of checking and calculating the whole delay and key.

Reply:
the sound is just quiet, turn off the background music and you can hear it, I thought that for walking the sound shouldn't be too laud

Reply:
the delay is bc if you press the key the music will overlap and it doesn't sound nice. I made it play every 10 step

Reply:
but I can try with the keydown maybe it will be better

Reply:
Well if you use paused and loop properties then it should be much easier and more efficient than this